server/core/src/https/mod.rs

```diff
@@ -429,568 +128,306 @@ pub async fn create_https_server(
             });
         }
     };
+    js_files
+}
 
-    let mut tserver = tide::Server::with_state(AppState {
+pub async fn create_https_server(
+    config: Configuration,
+    // trust_x_forward_for: bool, // TODO: #1787 make XFF headers work
+    cookie_key: [u8; 32],
+    jws_signer: JwsSigner,
+    status_ref: &'static StatusActor,
+    qe_w_ref: &'static QueryServerWriteV1,
+    qe_r_ref: &'static QueryServerReadV1,
+    mut rx: broadcast::Receiver<CoreAction>,
+) -> Result<tokio::task::JoinHandle<()>, ()> {
+    let jws_validator = jws_signer.get_validator().map_err(|e| {
+        error!(?e, "Failed to get jws validator");
+    })?;
+
+    let js_files = get_js_files(config.role);
+    // set up the CSP headers
+    // script-src 'self'
+    //      'sha384-Zao7ExRXVZOJobzS/uMp0P1jtJz3TTqJU4nYXkdmsjpiVD+/wcwCyX7FGqRIqvIz'
+    //      'sha384-MrcW6ZMFYlzcLA8Nl+NtUVF0sA7MsXsP1UyJoMp4YLEuNSfAP+JcXn/tWtIaxVXM'
+    //      'unsafe-eval';
+    let js_directives = js_files
+        .clone()
+        .into_iter()
+        .map(|f| f.hash)
+        .collect::<Vec<String>>();
+    let mut js_directives: Vec<CspValue> = js_directives
+        .into_iter()
+        .map(|value| CspValue::Sha384 { value })
+        .collect();
+    js_directives.extend(vec![CspValue::UnsafeEval, CspValue::SelfSite]);
+
+    let csp_header = axum_csp::CspSetBuilder::new()
+        // default-src 'self';
+        .add(CspDirectiveType::DefaultSrc, vec![CspValue::SelfSite])
+        // form-action https: 'self';
+        .add(
+            CspDirectiveType::FormAction,
+            vec![CspValue::SelfSite, CspValue::SchemeHttps],
+        )
+        // base-uri 'self';
+        .add(
+            CspDirectiveType::BaseUri,
+            vec![CspValue::SelfSite, CspValue::SchemeHttps],
+        )
+        // worker-src 'none';
+        .add(CspDirectiveType::WorkerSource, vec![CspValue::None])
+        // frame-ancestors 'none'
+        .add(CspDirectiveType::FrameAncestors, vec![CspValue::None])
+        .add(CspDirectiveType::ScriptSource, js_directives)
+        .add(
+            CspDirectiveType::ImgSrc,
+            vec![CspValue::SelfSite, CspValue::SchemeData],
+        );
+
+    // TODO this whole session store is kinda cursed and doesn't work the way we need, I think?
+    let store = async_session::CookieStore::new();
+    // let secret = b"..."; // MUST be at least 64 bytes!
```

Comment:
Is it 64 bytes or ... ?

Reply:
From the docs:
> `SessionLayer::new` will panic if the secret is less than 64 bytes.

And testing it, it did:

```
thread 'main' panicked at '`secret` must be at least 64 bytes.', /Users/yaleman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/axum-sessions-0.5.0/src/session.rs:107:13
```

Reply:
For now, I'd just make a vec and use the 32 bytes twice in a row. We'll fix it after.

Reply:
The current secret is 32 bytes, so we ... need to change it?